File: practic/src/main/java/com/group/practic/entity/Chapter.java
package com.group.practic.entity;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
public class Chapter {

	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE)
	Long id;
	
	@ManyToOne(fetch=FetchType.LAZY)
	Level level;

	@ManyToOne(fetch=FetchType.LAZY)
	Chapter parent;
	
	Integer step;

	String name;
	
	String refs;
}

File: frontend/src/main/angular/src/app/models/course/chapter.ts
export class Chapter {
  id: number;
  number: number;
  name: string;
  isVisible: boolean;

  constructor(id: number, number: number, name: string) {
    this.id = id;
    this.number = number;
    this.name = name;
    this.isVisible = false;
  }
}
